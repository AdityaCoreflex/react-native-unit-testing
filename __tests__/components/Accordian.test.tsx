import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Accordion from '../../src/components/Accordian';

describe('Accordion Component', () => {
  const props = {
    title: 'Sample Accordion',
    content: 'This is hidden content.',
    testID: 'accordion',
  };

  it('renders accordion header with title', () => {
    const { getByText } = render(<Accordion {...props} />);
    expect(getByText('Sample Accordion')).toBeTruthy();
  });

  it('does not show content initially', () => {
    const { queryByTestId } = render(<Accordion {...props} />);
    expect(queryByTestId('accordion-content')).toBeNull();
  });

  it('shows content on toggle press', () => {
    const { getByTestId, getByText } = render(<Accordion {...props} />);
    fireEvent.press(getByTestId('accordion-toggle'));
    expect(getByText('This is hidden content.')).toBeTruthy();
  });

  it('hides content again when toggled twice', () => {
    const { getByTestId, queryByTestId } = render(<Accordion {...props} />);
    fireEvent.press(getByTestId('accordion-toggle'));
    expect(queryByTestId('accordion-content')).toBeTruthy();

    fireEvent.press(getByTestId('accordion-toggle'));
    expect(queryByTestId('accordion-content')).toBeNull();
  });

  it('handles multiple accordion instances independently', () => {
    const { getByTestId, queryByTestId } = render(
      <>
        <Accordion title="Acc 1" content="Content 1" testID="acc1" />
        <Accordion title="Acc 2" content="Content 2" testID="acc2" />
      </>,
    );

    fireEvent.press(getByTestId('acc1-toggle'));
    expect(queryByTestId('acc1-content')).toBeTruthy();
    expect(queryByTestId('acc2-content')).toBeNull();

    fireEvent.press(getByTestId('acc2-toggle'));
    expect(queryByTestId('acc1-content')).toBeTruthy();
    expect(queryByTestId('acc2-content')).toBeTruthy();
  });
  it('matches snapshot (initially closed)', () => {
    const { toJSON } = render(<Accordion {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('matches snapshot after opening content', () => {
    const { getByTestId, toJSON } = render(<Accordion {...props} />);
    fireEvent.press(getByTestId('accordion-toggle'));
    expect(toJSON()).toMatchSnapshot();
  });

  it('matches snapshot after toggling open and closed', () => {
    const { getByTestId, toJSON } = render(<Accordion {...props} />);
    fireEvent.press(getByTestId('accordion-toggle'));
    fireEvent.press(getByTestId('accordion-toggle'));
    expect(toJSON()).toMatchSnapshot();
  });
});
