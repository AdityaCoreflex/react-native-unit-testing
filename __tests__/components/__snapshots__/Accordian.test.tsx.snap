// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component matches snapshot (initially closed) 1`] = `
<View
  testID="accordion"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#EEE",
        "borderRadius": 6,
        "marginVertical": 8,
        "opacity": 1,
        "padding": 12,
      }
    }
    testID="accordion-toggle"
  >
    <Text
      style={
        {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Sample Accordion
    </Text>
  </View>
</View>
`;

exports[`Accordion Component matches snapshot after opening content 1`] = `
<View
  testID="accordion"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#EEE",
        "borderRadius": 6,
        "marginVertical": 8,
        "opacity": 1,
        "padding": 12,
      }
    }
    testID="accordion-toggle"
  >
    <Text
      style={
        {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Sample Accordion
    </Text>
  </View>
  <View
    testID="accordion-content"
  >
    <Text
      style={
        {
          "backgroundColor": "#F5F5F5",
          "borderRadius": 6,
          "fontSize": 14,
          "padding": 12,
        }
      }
    >
      This is hidden content.
    </Text>
  </View>
</View>
`;

exports[`Accordion Component matches snapshot after toggling open and closed 1`] = `
<View
  testID="accordion"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#EEE",
        "borderRadius": 6,
        "marginVertical": 8,
        "opacity": 1,
        "padding": 12,
      }
    }
    testID="accordion-toggle"
  >
    <Text
      style={
        {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Sample Accordion
    </Text>
  </View>
</View>
`;
